common/src/main/java/me/shedaniel/architectury/event/events/GuiEvent.java

```diff
@@ -32,62 +32,131 @@
     }
     
     @Environment(EnvType.CLIENT)
     interface ScreenInitPre {
+        /**
+         * Called when a screen is initialized and after all widgets are cleared.
+         * Equal to the forge {@code GuiScreenEvent.InitGuiEvent.Pre} event.
+         * 
+         * @param screen The screen.
+         * @param widgets The widgets that are added after this event.
+         * @param children The listeners for the screen events.
+         * @return Returning {@link InteractionResult#FAIL} results in the rest of the init method being ignored.
+         */
         InteractionResult init(Screen screen, List<AbstractWidget> widgets, List<GuiEventListener> children);
     }
     
     @Environment(EnvType.CLIENT)
     interface ScreenInitPost {
+        /**
+         * Called when a screen is initialized after all the vanilla logic happened.
+         * Equal to the forge {@code GuiScreenEvent.InitGuiEvent.Post} event.
+         *
+         * @param screen The screen.
+         * @param widgets The widgets that were added.
+         * @param children The listeners for the screen events.
+         */
         void init(Screen screen, List<AbstractWidget> widgets, List<GuiEventListener> children);
     }
     
     @Environment(EnvType.CLIENT)
     interface ScreenRenderPre {
+        /**
+         * Called before any rendering is happening.
+         * Equal to the forge {@code GuiScreenEvent.DrawScreenEvent.Pre} event.
+         * 
+         * @param screen The screen.
+         * @param matrices The render buffer.
+         * @param mouseX The mouse x position.
+         * @param mouseY The mouse y position.
+         * @param delta The current tick delta.
+         * @return Returning {@link InteractionResult#FAIL} prevents any other rendering.
+         */
         InteractionResult render(Screen screen, PoseStack matrices, int mouseX, int mouseY, float delta);
     }
     
     @Environment(EnvType.CLIENT)
     interface ScreenRenderPost {
+        /**
+         * Called after all things have been rendered.
+         * Equal to the forge {@code GuiScreenEvent.DrawScreenEvent.Post} event.
+         *
+         * @param screen The screen.
+         * @param matrices The render buffer.
+         * @param mouseX The mouse x position.
+         * @param mouseY The mouse y position.
+         * @param delta The current tick delta.
+         */
         void render(Screen screen, PoseStack matrices, int mouseX, int mouseY, float delta);
     }
     
     @Environment(EnvType.CLIENT)
     interface SetScreen {
+        /**
+         * Called when a screen should be opened.
+         * Equal to the forge {@code GuiOpenEvent} event.
+         * 
+         * @param screen The screen that is gonna be opened.
+         * @return Return {@link InteractionResultHolder#fail(Object)} leads to the screen not being opened. The result is ignored. {@link InteractionResultHolder#success(Object)} leads to the passed new screen being used instead of the old one.
```

Comment:
```suggestion
         * @return Returning {@link InteractionResultHolder#fail(Object)} leads to the screen not being opened. The result is ignored. {@link InteractionResultHolder#success(Object)} leads to the passed new screen being used instead of the old one.
```